[Docs] README revamp (#980)
mlc-ai/mlc-llm: README.md

# MLC LLM

[Documentation](https://mlc.ai/mlc-llm/docs/) | [Blog](https://mlc.ai/blog/) | [WebLLM](https://webllm.mlc.ai/) | [WebStableDiffusion](https://websd.mlc.ai/) | [Discord][discord-url]

Machine Learning Compilation for Large Language Models (MLC LLM) is a high-performance **universal deployment** solution that allows native deployment of any large language models with native APIs with compiler acceleration.

> The mission of this project is to enable everyone to develop, optimize and deploy AI models natively on everyone's devices with ML compilation techniques.

Reviewer: Merge the two paragraphs

MLC LLM supports the following platforms and hardware:

<table style="width:100%">
  <thead>
    <tr>
      <th style="width:16%"> </th>
      <th style="width:21%">AMD GPU</th>
      <th style="width:21%">NVIDIA GPU</th>
      <th style="width:21%">Apple M1/M2 GPU</th>
      <th style="width:21%">Intel GPU</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Linux / Win</td>
      <td>✅ Vulkan, ROCm</td>
      <td>✅ Vulkan, CUDA</td>
      <td>N/A</td>
      <td>✅ Vulkan</td>
    </tr>
    <tr>
      <td>macOS</td>
      <td>✅ Metal</td>
      <td>N/A</td>
      <td>✅ Metal</td>
      <td>✅ Metal</td>
    </tr>
    <tr>
      <td>Web Browser</td>
      <td>✅ WebGPU</td>
      <td>✅ WebGPU</td>
      <td>✅ WebGPU</td>
      <td>✅ WebGPU</td>
    </tr>
    <tr>
      <td>iOS / iPadOS</td>
      <td colspan=4>✅ Metal on Apple M1/M2 GPU</td>
    </tr>
    <tr>
      <td>Android</td>
      <td colspan=2>✅ OpenCL on Adreno GPU</td>
      <td colspan=2>✅ OpenCL on Mali GPU</td>
    </tr>
  </tbody>
</table>

## News

* [08/02/2023] [Dockerfile](https://github.com/junrushao/llm-perf-bench/) released for CUDA performance benchmarking
* [07/19/2023] Supports 7B/13B/70B Llama-2

> ## Universal Deployment APIs

Reviewer: Move "Getting Started" before "Universal Deployment APIs"

MLC LLM provides multiple sets of APIs across platforms and environments, which you can use to build
your own LLM applications with native acceleration by MLC LLM. These include:

> * [Python API](https://mlc.ai/mlc-llm/docs/deploy/python.html),
> * [Rest API with OpenAI API compatibility](https://mlc.ai/mlc-llm/docs/deploy/rest.html),
> * [C++ API](https://mlc.ai/mlc-llm/docs/deploy/cli.html),
> * [JavaScript API](https://mlc.ai/mlc-llm/docs/deploy/javascript.html) and [Web LLM](https://github.com/mlc-ai/web-llm),
> * [Swift API for iOS App](https://mlc.ai/mlc-llm/docs/deploy/ios.html), and
> * [Java API and Android App](https://mlc.ai/mlc-llm/docs/deploy/android.html).

Reviewer: ```suggestion
* [Python API](https://mlc.ai/mlc-llm/docs/deploy/python.html)
* [OpenAI-Compatible Rest API](https://mlc.ai/mlc-llm/docs/deploy/rest.html)
* [C++ API](https://mlc.ai/mlc-llm/docs/deploy/cli.html)
* [JavaScript API](https://mlc.ai/mlc-llm/docs/deploy/javascript.html) and [Web LLM](https://github.com/mlc-ai/web-llm)
* [Swift API for iOS App](https://mlc.ai/mlc-llm/docs/deploy/ios.html)
* [Java API and Android App](https://mlc.ai/mlc-llm/docs/deploy/android.html)
```

> [Project Page](https://mlc.ai/mlc-llm/) | [Documentation](https://mlc.ai/mlc-llm/docs/) | [Blog](https://mlc.ai/blog/2023/05/01/bringing-accelerated-llm-to-consumer-hardware) | [WebLLM](https://webllm.mlc.ai/) | [WebStableDiffusion](https://websd.mlc.ai/) | [Discord][discord-url]

Reviewer: ```suggestion
[Project Page](https://mlc.ai/mlc-llm/) | [Documentation](https://mlc.ai/mlc-llm/docs/) | [Blog](https://blog.mlc.ai/) | [WebStableDiffusion](https://websd.mlc.ai/) | [Discord][discord-url]
```

Reviewer 2: let us remove project landing page link for now and instead first focus on the documentation. Then consider add it back later. The landing page will no longer host demos and will also contain info ptrs to docs

> MLC LLM is a high-performance **universal deployment** solution for large-language models that allows **any language models** to be **deployed natively** on a diverse set of hardware backends and native applications, plus a **productive framework** for everyone to further optimize model performance for their own use cases.

Reviewer: ```suggestion
Machine Learning Compilation for Large Language Models (MLC LLM) is a high-performance **universal deployment** solution that allows native deployment of any large language models with native APIs with compiler acceleration.
```

> The cornerstone of MLC LLM is machine learning compilation (MLC) and the mission of MLC LLM is to **enable everyone to develop, optimize and deploy AI models natively on everyone's devices**.

Reviewer: ```suggestion
The mission of the project is to enable everyone to develop, optimize and deploy AI models natively on everyone's devices with ML compilation techniques.
```

MLC LLM supports the following platforms and hardware:

<table style="width:100%">
  <thead>
    <tr>
      <th style="width:16%"> </th>
      <th style="width:21%">AMD GPU</th>
      <th style="width:21%">NVIDIA GPU</th>
      <th style="width:21%">Apple M1/M2 GPU</th>
      <th style="width:21%">Intel GPU</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Linux / Win</td>
      <td>✅ Vulkan, ROCm</td>
      <td>✅ Vulkan, CUDA</td>
      <td>N/A</td>
      <td>✅ Vulkan</td>
    </tr>
    <tr>
      <td>macOS</td>
      <td>✅ Metal</td>
      <td>N/A</td>
      <td>✅ Metal</td>
      <td>✅ Metal</td>
    </tr>
    <tr>
      <td>Web Browser</td>
      <td>✅ WebGPU</td>
      <td>✅ WebGPU</td>
      <td>✅ WebGPU</td>
      <td>✅ WebGPU</td>
    </tr>
    <tr>
      <td>iOS / iPadOS</td>
      <td colspan=4>✅ Metal on Apple M1/M2 GPU</td>
    </tr>
    <tr>
      <td>Android</td>
      <td colspan=2>✅ OpenCL on Adreno GPU</td>
      <td colspan=2>✅ OpenCL on Mali GPU</td>
    </tr>
  </tbody>
</table>

> MLC LLM provides multiple sets of APIs across platforms and environments, which you can use to build
> your own LLM applications with native acceleration by MLC LLM. These include:
> * [Python API and gradio frontend](https://mlc.ai/mlc-llm/docs/deploy/python.html),
> * [Rest API with OpenAI API compatibility](https://mlc.ai/mlc-llm/docs/deploy/rest.html),
> * [C++ API](https://mlc.ai/mlc-llm/docs/deploy/cli.html),
> * [JavaScript API](https://mlc.ai/mlc-llm/docs/deploy/javascript.html) and [Web LLM](https://github.com/mlc-ai/web-llm),
> * [Swift API for iOS](https://mlc.ai/mlc-llm/docs/deploy/ios.html), and
> * [Android App](https://mlc.ai/mlc-llm/docs/deploy/android.html).

Reviewer: ```suggestion

## Universal Deployment and APIs

* [Python API](https://mlc.ai/mlc-llm/docs/deploy/python.html);
* [Rest API with OpenAI API compatibility](https://mlc.ai/mlc-llm/docs/deploy/rest.html);
* [C++ API](https://mlc.ai/mlc-llm/docs/deploy/cli.html);
* [JavaScript API](https://mlc.ai/mlc-llm/docs/deploy/javascript.html) and [WebLLM](https://github.com/mlc-ai/web-llm);
* [Swift API and iOS App](https://mlc.ai/mlc-llm/docs/deploy/ios.html);
* [Java API and Android App](https://mlc.ai/mlc-llm/docs/deploy/android.html).
```

Author: I’m wondering if it is good to explicit mention that people can do secondary development with the API.

Reviewer: I do want to emphasize that our API surface is awesome awesome, but to be clear, those documents haven’t covered APIs yet, unfortunately. Let’s avoid overselling for now

> ## News

Reviewer: Move "News" before "Universal Deployment APIs"

* [08/02/2023] [Dockerfile](https://github.com/junrushao/llm-perf-bench/) released for CUDA performance benchmarking
* [07/19/2023] Supports 7B/13B/70B Llama-2

## Getting Started

<ins>**[Check out our instruction page to try out!](https://mlc.ai/mlc-llm/docs/get_started/try_out.html)**</ins>

## Citation

> If you find MLC LLM useful in your work, please consider citing the project using the following format:

Reviewer: ```suggestion
Please consider citing our project if you find it useful:
```

```bibtex
@software{mlc-llm,
    author = {MLC team},
    title = {{MLC-LLM}},
    url = {https://github.com/mlc-ai/mlc-llm},
    year = {2023}
}
```

> The underlying compiler techniques employed by MLC LLM are outlined in the following papers:

Reviewer: ```suggestion
The underlying compiler techniques of MLC LLM include:
```

      booktitle = {Advances in Neural Information Processing Systems},
      editor = {S. Koyejo and S. Mohamed and A. Agarwal and D. Belgrave and K. Cho and A. Oh},
      pages = {35783--35796},
      publisher = {Curran Associates, Inc.},
      title = {Tensor Program Optimization with Probabilistic Programs},
      url = {https://proceedings.neurips.cc/paper_files/paper/2022/file/e894eafae43e68b4c8dfdacf742bcbf3-Paper-Conference.pdf},
      volume = {35},
      year = {2022}
  }

  @inproceedings{tvm,
      author = {Tianqi Chen and Thierry Moreau and Ziheng Jiang and Lianmin Zheng and Eddie Yan and Haichen Shen and Meghan Cowan and Leyuan Wang and Yuwei Hu and Luis Ceze and Carlos Guestrin and Arvind Krishnamurthy},
      title = {{TVM}: An Automated {End-to-End} Optimizing Compiler for Deep Learning},
      booktitle = {13th USENIX Symposium on Operating Systems Design and Implementation (OSDI 18)},
      year = {2018},
      isbn = {978-1-939133-08-3},
      address = {Carlsbad, CA},
      pages = {578--594},
      url = {https://www.usenix.org/conference/osdi18/presentation/chen},
      publisher = {USENIX Association},
      month = oct,
  }
  ```
</details>

> ## Links

Reviewer: Remove "WebLLM", "project pag", "iOS app" in "Links" section
